src/lib.mo

```diff
@@ -70,23 +65,24 @@ module {
 
       state := #ready;
       result := post(midstate);
+      if (debug_) Debug.print("Response is #ready. " # debug_message);
 
-      if (Option.isNull(result)) throw Error.reject("Reponse rejected. " # debug_message);
+      if (Option.isNull(result)) throw Error.reject("Response rejected. " # debug_message);
     };
   };
 
-  class BaseTester<T, S, R>(name : ?Text, iterations_limit : ?Nat) {
+  class BaseTester<T, S, R>(name : ?Text, iterations_limit : ?Nat, debug_ : Bool) {
     var queue : Buffer.Buffer<Response<T, S, R>> = Buffer.Buffer(1);
     public var front = 0;
     let limit = Option.get(iterations_limit, 100);
 
-    public func call_result(i : Nat) : R {
-      let ?r = get(i).result else Debug.trap("No call result");
+    public func call_result(index : Nat) : R {
+      let ?r = get(index).result else Debug.trap("No call result");
       r;
     };
 
-    public func add(lock : Bool, pre : PreFunc<T, S>, post : PostFunc<S, R>, key : ?Text) {
-      let response = Response<T, S, R>(lock, pre, post, name, key, limit);
+    public func add(lock : Bool, pre : PreFunc<T, S>, post : PostFunc<S, R>) {
+      let response = Response<T, S, R>(lock, pre, post, debug_, name, queue.size(), limit);

```

Comment:
Here I would pass in `debugMessage(name, queue.size())` instead of the two values individually.